### PR TITLE
docs(config): add new option `workerChunkLoading` for output

### LINKED ISSUE #3960: Document webpack change: new Worker() support
*A pull request by @sokra was merged and maintainers requested a documentation change.*

See pull request: https://github.com/webpack/webpack/pull/11439

---

* add support for async anonymous entrypoints from inside the codebase
* add worker example
* move entry options into Entrypoint and receive them from there

This adds support to create entrypoints from AsyncDependenciesBlocks. Might also be useful for other scenarios (@timneutkens `interactive`)

**What kind of change does this PR introduce?**
feature

**Did you add tests for your changes?**
yes

**Does this PR introduce a breaking change?**
yes

**What needs to be documented once your changes are merged?**
* new option `ouput.workerChunkLoading` controls the chunk loading of workers (default chosen based on `output.chunkLoading`)
* some syntax triggers webpack to compile "..." as new entrypoint with `ouput.workerChunkLoading` as chunk loading mechnism
  * `new Worker(new URL("...", import.meta.url), ...)` with optional `import { Worker } from "worker_threads"`
  * `navigator.serviceWorker.register(new URL("...", import.meta.url), ...) `

### PR DESCRIPTION
**Pull request by QC-L**
Resolve #3960

### CODE REVIEW

File: src/content/configuration/output.md [hunk 1]
```diff
@@ -1133,6 +1134,25 @@ module.exports = {
 };
 ```
 
+## `output.workerChunkLoading`
+
+`string = 'require' | 'import-scripts' | 'async-node' | 'import' | 'universal'` `boolean: false` 
+
+The new option `workerChunkLoading` controls the chunk loading of workers. 
+
+T> The default value of this option changes when `target` is different. For more details, see [`output.workerChunkLoading` default value](https://github.com/webpack/webpack/blob/master/lib/config/defaults.js#L604) and [`target.js`](https://github.com/webpack/webpack/blob/master/lib/config/target.js#L64).
```
Comment (EugeneHlushko) [inline src/content/configuration/output.md:1143]: We have faced numerous issues with linking to LOC on webpack master, code moves too quick. Suggest we note the searchable option in a specified file instead
```suggestion
T> The default value of this option is depending on the `target` setting. For more details, search for `"workerChunkLoading"` : [ in the webpack defaults](https://github.com/webpack/webpack/blob/master/lib/config/defaults.js).
```

**Review by EugeneHlushko — CHANGES_REQUESTED** [review]

File: src/content/configuration/output.md (hunk 1 above)
Comment (QC-L) [reply src/content/configuration/output.md:1143]: Okay. I see.

Comment (QC-L) [reply src/content/configuration/output.md:1143]:
```suggestion
T> The default value of this option is depending on the `target` setting. For more details, search for `"workerChunkLoading"`: [in the webpack defaults](https://github.com/webpack/webpack/blob/master/lib/config/defaults.js).
```

**Comment by QC-L** [comment]
@EugeneHlushko Done.

**Comment by chenxsan** [comment]
CI has been fixed in the master branch, you might update your `master` branch first and merge the changes into this branch with `git merge master`.

**Comment by EugeneHlushko** [comment]
it wasnt completely fixed. As after we've removed deprecated eslint-loader, one of the plugins pages was trying to link to its local destination under webpack.js.org instead of directly to github, i've fixed in in #4013

**Review by EugeneHlushko — APPROVED** [review]

**Comment by EugeneHlushko** [comment]
Thanks!

**Pull request merged**